extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -479,6 +485,33 @@ public void handle() throws InterruptedException, ExecutionException
     }
   }
 
+  private class ResetNotice implements Notice
+  {
+    @Override
+    public void handle()
+    {
+      resetInternal();
+    }
+  }
+
+  @VisibleForTesting
+  void resetInternal()
+  {
+    boolean result = indexerMetadataStorageCoordinator.deleteDataSourceMetadata(dataSource);
+    log.info("Reset dataSource[%s] - dataSource metadata entry deleted? [%s]", dataSource, result);
+
+    for (TaskGroup taskGroup : taskGroups.values()) {
+      for (Map.Entry<String, TaskData> entry : taskGroup.tasks.entrySet()) {
+        String taskId = entry.getKey();
+        log.info("Reset dataSource[%s] - killing task [%s]", dataSource, taskId);
+        killTask(taskId);
+      }
+    }
+
+    partitionGroups.clear();
+    taskGroups.clear();
```

Comment:
@pjain1 it shouldn't be necessary; when the tasks are killed they will fire events that will post a RunNotice on line 435 + the scheduled execution will still be active and will trigger the run loop